test/batch_restart_pods_test.go

```diff
@@ -565,3 +494,92 @@ func schedulableResource(mem string) *corev1.ResourceRequirements {
 		},
 	}
 }
+
+func batchRollingRestartTest(
+	k8sClient client.Client, ctx goctx.Context,
+	clusterNamespacedName types.NamespacedName,
+	batchSize *intstr.IntOrString,
+) error {
+	aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+	if err != nil {
+		return err
+	}
+
+	aeroCluster.Spec.RackConfig.RollingUpdateBatchSize = batchSize
+	aeroCluster.Spec.PodSpec.AerospikeContainerSpec.Resources = unschedulableResource()
+
+	err = updateClusterForBatchRestart(k8sClient, ctx, aeroCluster)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func batchUpgradeTest(
+	k8sClient client.Client, ctx goctx.Context,
+	clusterNamespacedName types.NamespacedName,
+	batchSize *intstr.IntOrString,
+) error {
+	aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+	if err != nil {
+		return err
+	}
+
+	aeroCluster.Spec.RackConfig.RollingUpdateBatchSize = batchSize
+	aeroCluster.Spec.Image = unavailableImage
+
+	err = updateClusterForBatchRestart(k8sClient, ctx, aeroCluster)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func rollingRestartTest(
+	k8sClient client.Client, ctx goctx.Context,
+	clusterNamespacedName types.NamespacedName,
+	batchSize *intstr.IntOrString, mem string,
+) error {
+	aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+	if err != nil {
+		return err
+	}
+
+	aeroCluster.Spec.RackConfig.RollingUpdateBatchSize = batchSize
+
+	if mem != "" {
+		aeroCluster.Spec.PodSpec.AerospikeContainerSpec.Resources = schedulableResource(mem)
+	} else {
+		aeroCluster.Spec.PodSpec.AerospikeContainerSpec.Resources = unschedulableResource()
+	}
+
+	err = updateCluster(k8sClient, ctx, aeroCluster)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func upgradeTest(
+	k8sClient client.Client, ctx goctx.Context,
+	clusterNamespacedName types.NamespacedName,
+	batchSize *intstr.IntOrString, image string,
+) error {
+	aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+	if err != nil {
+		return err
+	}
+
+	aeroCluster.Spec.RackConfig.RollingUpdateBatchSize = batchSize
+	aeroCluster.Spec.Image = image
+
+	err = updateCluster(k8sClient, ctx, aeroCluster)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
In all these funcs, we can directly return the error var at the end instead of checking